lib/open_api_spex/schema.ex

```diff
@@ -185,12 +185,16 @@ defmodule OpenApiSpex.Schema do
     {:error, "Invalid array: #{inspect(value)}"}
   end
   def cast(schema = %Schema{type: :object}, value, schemas) when is_map(value) do
+    schema = %{schema | properties: schema.properties || %{}}
```

Comment:
I'm not sure that this approach of checking for the `schema.properties` value in various different places is a good idea since this check can be easily missed in some other places which will eventually result in an error.

Having a reasonable default value (`%{}` in this case for `properties`) seems a better way to go, but i'm fine with either way as long as all the tests pass.